### PR TITLE
Replace use of deprecated 'imp' module by 'importlib'

### PR DESCRIPTION
**Pull request by mthuurne**
This was extracted from a drive-by fix in #241.

### CODE REVIEW

**Comment by mthuurne** [comment]
@adiroiban: Your idea to use `cython-test-exception-raiser` to test introspection of extensions works. I did have to add a workaround to make it pass on pypy3 as well.

File: pydoctor/test/test_model.py [hunk 1]
```diff
@@ -166,19 +165,45 @@ class Dummy:
...
     assert method.docstring == "Mmm"
+
+
+def test_introspection_extension() -> None:
+    """Find docstrings from this test using introspection of an extension."""
+
+    try:
+        import cython_test_exception_raiser.raiser
+    except ImportError:
+        pytest.skip("cython_test_exception_raiser not installed")
```
Comment (adiroiban) [inline pydoctor/test/test_model.py:191]: I guess that we can assume that the extension is always installed and don't handle the import error.

We have wheels for major OSes so it should be easy to install it.

**Review by adiroiban — APPROVED** [review]
Wow! Excellent work.

Happy to see it merged.

Thanks

File: pydoctor/test/test_model.py (hunk 1 above)
Comment (mthuurne) [reply pydoctor/test/test_model.py:191]: I thought pypy might need it, but it seems that the extension is available in pypy too.

Still, I think it's good to be able to run the tests on any platform, even if the extension is not available or the user hasn't installed it.

Comment (adiroiban) [reply pydoctor/test/test_model.py:191]: my concern is to not end up with accidentally not installing the extension and then the test is always skipped... like we had with twisted tests and pydoctor not being installed.

To mitigate this, I am using codecov and a separate target that makes sure the tests code have 100%  coverage... if someone some tests are always skipped then the coverage will no longer be 100% and end up with a red flag :)

Comment (mthuurne) [reply pydoctor/test/test_model.py:191]: That's a good idea. I want to work on other things right now though, so I filed #245 to make sure we remember it.

**Pull request merged**